test/manifold_test.cpp

```diff
@@ -784,3 +784,56 @@ TEST(Manifold, EmptyHull) {
       {0, 0, 0}, {1, 0, 0}, {0, 1, 0}, {1, 1, 0}};
   EXPECT_TRUE(Manifold::Hull(coplanar).IsEmpty());
 }
+
+TEST(Manifold, ConvexConvexMinkowski) {
+  bool oldDeterministic = ManifoldParams().deterministic;
+  ManifoldParams().deterministic = true;
```

Comment:
What difference did you find this made? I've been seeing some surprising results between turning this on and off. @pca006132 I can't remember what the tradeoffs are anymore - is there any reason we would ever want to be not deterministic?

Reply:
I'm a bit paranoid about the non-determinism of the genus etc. that arise from non-determinism in the unit tests... and I think disabling determinism enables some parallel optimizations, which result in the booleans getting added in a random order...

At least, until the precision behavior is found/fixed, it's less stressful on me for the unit tests to have these 😅

Reply:
Yes, this disables the optimization that allows boolean operations to race against each other, so the result is deterministic (under the exact same condition and under the same binary). I think for reproducing results in the CI, it is fine to force it to be deterministic, but in general this may indicate that there are bugs or the test is testing properties that are too fragile and may break in future modification.

Reply:
Hrm, I'm trying some more robust normal calculation methods for the offset PR, and I'm noticing that the rounding results are still non-deterministic between runs, despite setting the determinism parameter...

In my case, 1-in-10 runs of the test negative offset I'm doing come out without any slivers or genus issues; these runs are marked by executing more quickly (~25%) than the others.   Thus, I think whatever non-determinism that is affecting kissing triangles is outside the purview of `ManifoldParams().deterministic = true;`.

Reply:
ah, we should make normal calculation sequential if we need determinism.

Reply:
Where does it happen?   I'm having trouble finding it...

Reply:
Aha, I find that if I make add the determinism check to the two policies on these lines, I get deterministic slivers and execution times:
https://github.com/elalish/manifold/blob/7d564cfbc19ee606523cbabb1c34d96fb748b9c6/src/manifold/src/csg_tree.cpp#L226-L230

The real solution for slivers will probably be to make sure that each vertex is extruding along the average of the co-planar faces it's surrounded by.
That way, it will always end up at the same point as the equivalent vertex on the neighboring triangle 🤔

Reply:
Yeah, I put some effort into making the Boolean not recalculate face normals, but copy them to any triangles that came from them for exactly this reason. However, with all the use of hulls, that may be defeated. Still, my Warp method avoided that pitfall, but still had the problem, so there may be other things going on too.

Reply:
Doing as much of the computation as possible in `double` space seems to ~halve the floaters https://github.com/zalo/manifold/commit/25f395892e00593d4e346942f2f6fde8475037bf 😅 

Unfortunately, it doesn't seem like `glm` has facilities for `long double` math 🙃